lib/engine-dylib/src/artifact.rs

```diff
@@ -402,6 +402,7 @@ impl DylibArtifact {
 
         let notext = match (target_triple.operating_system, target_triple.architecture) {
             (OperatingSystem::Linux, Architecture::X86_64) => vec!["-Wl,-z,notext"],
+            (OperatingSystem::MacOSX, Architecture::X86_64) => vec!["-Wl,-read_only_relocs,suppress"],
```

Comment:
We actually want to remove both of these options. They currently tell the linker to allow text relocations which we *don't* want for PIC code.